src/frontend/js/components/LtiConsumer/index.tsx

```diff
@@ -1,47 +1,63 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { iframeResizer } from 'iframe-resizer';
-import { LtiConsumer as LtiConsumerProps } from 'types/LtiConsumer';
+import { LtiConsumerContext, LtiConsumerProps } from 'types/LtiConsumer';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+import { handle } from 'utils/errors/handle';
 
-const LtiConsumer = ({
-  url,
-  content_parameters: contentParameters,
-  automatic_resizing,
-}: LtiConsumerProps) => {
+const LtiConsumer = ({ id }: LtiConsumerProps) => {
   const formRef = React.useRef<HTMLFormElement>(null);
   const iframeRef = React.useRef<HTMLIFrameElement>(null);
+  const [context, setContext] = useState<LtiConsumerContext>();
 
-  useEffect(() => {
-    formRef.current?.submit();
-    if (automatic_resizing) {
-      // Retrieve and inject current component container height to prevent flickering
-      // and remove aspect-ratio trick which is not compatible with iframeResizer
-      const componentContainer = document.querySelector('.richie-react--lti-consumer');
-      iframeResizer({ minHeight: componentContainer?.clientHeight }, iframeRef.current!);
-      componentContainer?.classList.remove('aspect-ratio');
-      componentContainer?.attributes.removeNamedItem('style');
+  const checkResponseStatus = (response: Response) => {
+    if (!response.ok) {
+      throw new Error(`Failed to retrieve LTI consumer context at placeholder ${id}`);
     }
+    return response.json();
+  };
+
+  useAsyncEffect(async () => {
+    await fetch(`/api/v1.0/plugins/lti-consumer/${id}/context/`)
```

Comment:
I don't get how toolbar.edit_mode_active is added to the call here…
Is it managed natively by django ?

Reply:
There is `ToolbarMiddleware` which is in charge to check if request `is_cms_request` then to bind the toolbar to the request.

https://github.com/django-cms/django-cms/blob/6e5caadc30748eb944129c950f668292aae9a348/cms/middleware/toolbar.py#L18